app/lib/core/utilities/reponse_manager.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:app/core/utilities/optional.dart';
+import 'package:flutter/material.dart';
+
+enum ResponseType { success, failure, catastrophic_failure }
```

Comment:
- [ ] We should still drop the `catastrophic_failure` and just use success and failure.

So, you should also change `Response.failure` to set the type as `failure` by default like what we did in our last discussion.

Reply:
Will do, thanks